Lib/contextlib.py

```diff
@@ -197,6 +216,7 @@ def __exit__(self, typ, value, traceback):
             try:
                 raise RuntimeError("generator didn't stop after throw()")
             finally:
+                del self.exc_context
```

Comment:
Likewise:

```suggestion
                self.exc_context = None
```

or, better:

```py
    def __exit__(self, typ, value, traceback):
        exc_context = self.exc_context
        self.exc_context = None
```